addons/analytic/models/analytic_mixin.py

```diff
@@ -13,6 +13,11 @@ class AnalyticMixin(models.AbstractModel):
         compute="_compute_analytic_distribution", store=True, copy=True, readonly=False,
         precompute=True
     )
+    # Json non stored to be able to search on analytic_distribution.
+    analytic_distribution_search = fields.Json(
```

Comment:
Can't you add the search function on the `analytic_distribution` field?

Reply:
No, we can't add a search function on a stored field (https://github.com/odoo/odoo/blob/16.0/odoo/osv/expression.py#L731)

Reply:
The ORM team already looked into adding that feature, but they asked us to do it like this because "it is too complex/risky" to do.